engine/classes/Elgg/I18n/Translator.php

```diff
@@ -142,33 +159,63 @@ public function translate($message_key, array $args = [], $language = "") {
 		$langs['en'] = true;
 
 		// try to translate
-		$notice = '';
 		$string = $message_key;
+
+		// separate into sprintf_args and kwargs
+		list($printf_args, $kwargs) = self::getStringReplacementsArgs($args);
```

Comment:
We don't want to encourage object method usage, so we should just flatten `$kwargs` here using `ElggData::toObject()`